src/lib/services/api-token-service.ts

```diff
@@ -92,6 +106,19 @@ export class ApiTokenService {
         }
     }
 
+    async updateLastSeen(): Promise<void> {
+        if (this.lastSeenSecrets.size > 0) {
+            let toStore = new Set<string>();
+            [this.lastSeenSecrets, toStore] = [toStore, this.lastSeenSecrets];
+            await this.store.markSeenAt([...toStore]);
```

Comment:
This syntax makes me a bit nervous, as it requires the next line to be a spread operator. Remember that node processes are single threaded, so we can have two lines in a row and assume order (as long as they are not async). 

**How about:**

```suggestion
            let toStore = [...this.lastSeenSecrets];
            this.lastSeenSecrets = new Set<string>();
            await this.store.markSeenAt(toStore);
```